modules/tinymce/src/themes/silver/main/ts/ui/toolbar/button/ToolbarButtons.ts

```diff
@@ -168,27 +186,30 @@ const renderFloatingToolbarButton = (spec: Toolbar.GroupToolbarButton, backstage
 
 const renderCommonToolbarButton = <T>(spec: GeneralToolbarButton<T>, specialisation: Specialisation<T>, providersBackstage: UiFactoryBackstageProviders): SketchSpec => {
   const editorOffCell = Cell(Fun.noop);
-  const structure = renderCommonStructure(spec.icon, spec.text, spec.tooltip, Optional.none(), Optional.none(), providersBackstage);
+  const structure = renderCommonStructure(spec.icon, spec.text, spec.tooltip, Optional.none(), providersBackstage);
   return AlloyButton.sketch({
     dom: structure.dom,
     components: structure.components,
 
     eventOrder: toolbarButtonEventOrder,
-    buttonBehaviours: Behaviour.derive(
-      [
-        AddEventsBehaviour.config('toolbar-button-events', [
-          onToolbarButtonExecute<T>({
-            onAction: spec.onAction,
-            getApi: specialisation.getApi
-          }),
-          onControlAttached(specialisation, editorOffCell),
-          onControlDetached(specialisation, editorOffCell)
-        ]),
-        // Enable toolbar buttons by default
-        DisablingConfigs.toolbarButton(() => !spec.enabled || providersBackstage.isDisabled()),
-        ReadOnly.receivingConfig()
-      ].concat(specialisation.toolbarButtonBehaviours)
-    )
+    buttonBehaviours: {
+      ...Behaviour.derive(
+        [
+          AddEventsBehaviour.config('toolbar-button-events', [
+            onToolbarButtonExecute<T>({
+              onAction: spec.onAction,
+              getApi: specialisation.getApi
+            }),
+            onControlAttached(specialisation, editorOffCell),
+            onControlDetached(specialisation, editorOffCell)
+          ]),
+          // Enable toolbar buttons by default
+          DisablingConfigs.toolbarButton(() => !spec.enabled || providersBackstage.isDisabled()),
+          ReadOnly.receivingConfig()
+        ].concat(specialisation.toolbarButtonBehaviours)
+      ),
+      [commonButtonDisplayEvent]: structure.buttonBehaviours?.[commonButtonDisplayEvent],
```

Comment:
This is a red flag for me - behaviours should always go inside the `derive` call. Not doing that means there's probably a bug lurking somewhere.

Reply:
That wasn't clear - please add a comment about why we're doing it this way, because (as far as I can tell) it's different to every other behaviour specification we have.

Reply:
Done!